security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -414,16 +412,15 @@ func (sc *SecretManagerClient) generateRootCertFromExistingFile(rootCertPath, re
 func (sc *SecretManagerClient) generateKeyCertFromExistingFiles(certChainPath, keyPath, resourceName string) (*security.SecretItem, error) {
 	// There is a remote possibility that key is written and cert is not written yet.
 	// To handle that case, check if cert and key are valid if they are valid then only send to proxy.
-	b := backoff.NewExponentialBackOff()
-	b.InitialInterval = sc.configOptions.FileDebounceDuration
+	b := backoff.NewExponentialBackOff(func(off *backoff.ExponentialBackOff) {
+		off.InitialInterval = sc.configOptions.FileDebounceDuration
+		off.MaxElapsedTime = totalTimeout
```

Comment:
I think with recent changes MaxElapsedTime is no longer a timeout. We also didn't have a timeout before... do we want to be adding one here?

Reply:
yes

Reply:
TestWorkloadAgentGenerateSecretFromFileOverSdsWithBogusFiles failed with no timeout